### PR TITLE
bazel: generate generic interval btree code in the bazel sandbox

### LINKED ISSUE #64166: bazel: _interval_btree code should be generated in the Bazel sandbox
This includes:

* pkg/kv/kvserver/spanlatch/latch_interval_btree.go
* pkg/kv/kvserver/spanlatch/latch_interval_btree_test.go
* pkg/kv/kvserver/concurrency/lockstate_interval_btree.go
* pkg/kv/kvserver/concurrency/lockstate_interval_btree_test.go

The `pkg/util/internal` directory also has some example generated files.

### PR DESCRIPTION
**Pull request by rickystewart**
We used to be using the checked-in versions of this generated code,
which of course there is no reason to be doing.

Resolves #64166

Release note: None

### CODE REVIEW

**Review by rail — APPROVED** [review]
:lgtm:

> Reviewed 7 of 7 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/64221#-:-MZG4mYV8JTpWjjFczZY:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rickystewart)

**Comment by rickystewart** [comment]
bors r=rail

File: pkg/kv/kvserver/concurrency/BUILD.bazel [hunk 1]
```diff
@@ -8,7 +9,7 @@ go_library(
         "latch_manager.go",
         "lock_table.go",
         "lock_table_waiter.go",
-        "lockstate_interval_btree.go",
+        ":lockstate_interval_btree.go",  # keep
```
Comment (nvanbenschoten) [inline pkg/kv/kvserver/concurrency/BUILD.bazel:12]: Out of curiosity, why did you add the colon up front?

Also, are all `# keep` comments manually written, or are they ever generated by gazelle?

Comment (rickystewart) [reply pkg/kv/kvserver/concurrency/BUILD.bazel:12]: I don't think the colon up front is strictly necessary, but to my eye it makes it more clear the file is generated inside this package rather than being a source file (like all the other files in this list).

The `# keep` comments themselves are generally manually written. As a rule they're added because otherwise Gazelle would delete the line in question.
